Remove LINQ and reflection from EasingTypeConverter

imememani opened the pull request:
### Description of Change

Clean up (using switch in place of if chain) for better readability and used more up to date .NET syntax as well as removing the usage of LINQ.

### Issues Fixed

N/A

Fixes #

N/A

imememani committed:
Clean up / Linq removal

Cleaned up the methods and added more up to date .NET syntax as well as removing the usage of LINQ.

imememani committed:
Removed ToLowerInvariant

As the switch makes use of nameof the usage of ToLowerInvariant is redundant.

affableashish commented:
[Switch expressions](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression) are much cleaner IMHO.

I recommend changing your switch statements to:

```c#
return strValue switch
{
    _ when Compare(nameof(Linear)) => Linear,
    _ when Compare(nameof(SinIn)) => SinIn,
    //  and so on...
    _ => "put default value here"
};
```

And add this local method for case insensitive comparison:
```c#
bool Compare(string right) => string.Equals(strValue, right, StringComparison.InvariantCultureIgnoreCase);
```

imememani committed:
Fix return value / further improvements

The value now returns an easing instead of a string (oops!), I have also implemented a local method to check case sensitive strings for the switch statement allowing the same functionality as previous.

imememani commented on src/Core/src/Converters/EasingTypeConverter.cs:
```diff
@@ -25,40 +24,50 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
+			switch (easing)
+			{
+				case nameof(Linear):
+					return nameof(Linear);
```
You are 100% correct, I've fixed the return value.

imememani commented on src/Core/src/Converters/EasingTypeConverter.cs:
```diff
@@ -25,40 +24,50 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
-				.DeclaredFields
-				.FirstOrDefault(f => f.Name.Equals(strValue, StringComparison.OrdinalIgnoreCase))
-				?.GetValue(null);
+			switch (easing)
```
Fixed, it should now take this into account, thanks!

imememani commented:
@affableashish Great suggestion, it does look much more readable and allows case to be ignored, I've pushed some changes to factor the suggestions in.

albyrock87 commented:
Is there a reason to not use `Enum.TryParse` (with case-insensitive bool parameter) and `ToString` instead of the two switch statements?

mattleibow commented:
/rebase

mattleibow committed:
Merge branch 'main' into pr/19260

mattleibow committed:
Tests

mattleibow committed:
Update EasingTypeConverter.cs

mattleibow commented:
/azp run
